.generator/src/generator/templates/api.j2

```diff
@@ -56,11 +56,7 @@ func (r *{{ operation.operationId }}OptionalParameters) With{{ name|camel_case }
 }
 {%- endfor %}
 
-/*
- * {{ operation.operationId }} {{ operation.summary }}
-{{ operation.description|block_comment(prefix=" *") }}
- */
-func (a *{{ classname }}Service) {{ operation.operationId }}(ctx _context.Context{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}} {{ get_type_for_parameter(parameter) }}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o ...{{ operation.operationId }}OptionalParameters{% endif %}{% endfor %}) ({% if returnType %}{{ returnType }}, {% endif %}*_nethttp.Response, error) {
+func (a *{{ classname }}Service) build{{ operation.operationId }}Request(ctx _context.Context{% for name, parameter in operation|parameters if parameter.required or parameter.in == "path" %}, {{ name|variable_name}} {{ get_type_for_parameter(parameter) }}{% endfor %}{%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}{%- if loop.first %}, o ...{{ operation.operationId }}OptionalParameters{% endif %}{% endfor %}) (api{{ operation.operationId }}Request, error) {
```

Comment:
Do you actually need the build helper? It looks like the pagination method ends up calling build/execute the same way as the the main function, no?

Reply:
It returns a reusable request object that could be updated instead of rebuilding it for each page. I would propose to do such optimization in a separate PR.